### PR TITLE
Marking the UI-metadata and Metrics artifacts as optional

### PR DESCRIPTION
**Pull request by Ark-kun**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1260)

### CODE REVIEW

File: sdk/python/kfp/compiler/_op_to_template.py [hunk 1]
```diff
@@ -193,6 +193,10 @@ def _op_to_template(op: BaseOp):
             for name, path in output_artifact_paths.items()
         ]
 
+        for output_artifact in output_artifacts:
+            if output_artifact['name'] in ['mlpipeline-ui-metadata', 'mlpipeline-metrics']:
+                output_artifact['optional'] = True
```
Comment (hongye-sun) [inline sdk/python/kfp/compiler/_op_to_template.py]: Is this field working with current argo image?

Comment (Ark-kun) [reply sdk/python/kfp/compiler/_op_to_template.py]: Yes =)
I've discovered that Argo is using slightly broken way to parse YAML, so unknown fields are ignored.
Let's see what the tests say.

Comment (Ark-kun) [reply sdk/python/kfp/compiler/_op_to_template.py]: All tests have passed.

Comment (hongye-sun) [reply sdk/python/kfp/compiler/_op_to_template.py]: I don't see the sample test passing for the last commit. Actually, it seems that it didn't run for the last commit. Do you know why?

Comment (Ark-kun) [reply sdk/python/kfp/compiler/_op_to_template.py]: It did. https://prow.k8s.io/pr-history/?org=kubeflow&repo=pipelines&pr=1260
But the unfinished tests are often shown at the rightmost column under the "Unknown" commit label.

Comment (hongye-sun) [reply sdk/python/kfp/compiler/_op_to_template.py]: sg. thanks.

/lgtm

Comment (Ark-kun) [reply sdk/python/kfp/compiler/_op_to_template.py]: // When I wrote the comment that the tests have passed, the sample test has already completed the main part and was deleting the cluster. I guess, when you've looked at it, the Prow still has not detected it as finished.

/approve

**Comment by gaoning777** [comment]
Add a argo PR link: https://github.com/argoproj/argo/pull/1277 as reference.

**Comment by Ark-kun** [comment]
> Add a argo PR link: [argoproj/argo#1277](https://github.com/argoproj/argo/pull/1277) as reference.

Done.

**Pull request merged**